httpie/output/formatters/json.py

```diff
@@ -1,14 +1,70 @@
 import json
+from typing import Generator
 
 from ...plugins import FormatterPlugin
 
+XXSI_PREFIXES = (")]}',\n", ")]}',", 'while(1);', 'for(;;)')
+
 
 class JSONFormatter(FormatterPlugin):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.enabled = self.format_options['json']['format']
 
+    def _find_potential_json_bodies(self, data: str) -> Generator[str, None, None]:
+        """Yield possible JSON bodies derived from `data`.
+
+        Sometimes the JSON body is prefixed with a XSSI token, specific to the server.
+        In order to be able to display well-formatted JSON, that method will try
+        several tricks to find the actual body of the JSON.
+
+        Here are yielded bodies (if one is successfully parsed by :meth:`._load_json()`,
+        then the process is stopped early):
+
+            1. First, the full `data`.
+            2. Then, strip off eventual :data::`XSSI_PREFIXES`.
+            3. Then, try to find the start of the actual body by applying some heuristics.
+            4. Finally, iterate over `data` one character at a time.
+               Note: that path should be rarely taken, it likely means
+                     `data` is a simple JSON string prefixed with a
+                     custom XSSI prefix, or not JSON at all.
+
+        """
+        # First, the full data.
+        yield data
+
+        # Then, strip off eventual XSSI prefix.
+        for prefix in XXSI_PREFIXES:
+            if data.startswith(prefix):
+                yield data[len(prefix)]
+
+        # Then, try to find the start of the actual body.
+        token_position = {data.find('{'), data.find('[')} - {-1}
+        if token_position:
+            yield data[min(token_position):]
+
+        # Finally, iterate over the data one character at a time.
+        # Note: that path should be rarely taken, it likely means
+        #       the data is a simple JSON string prefixed with a
+        #       custom XSSI prefix, or not JSON at all.
+        data = data[1:]
+        while data:
+            yield data
+            data = data[1:]
```

Comment:
This may be insane to do that. It is present only to handle simple JSON strings prefixed with custom XSSI string.